lib/emque/consuming/adapters/rabbit_mq/worker.rb

```diff
@@ -73,11 +78,48 @@ def process_message(delivery_info, metadata, payload)
               )
               ::Emque::Consuming::Consumer.new.consume(:process, message)
               channel.ack(delivery_info.delivery_tag)
-            rescue StandardError
-              channel.nack(delivery_info.delivery_tag)
+            rescue StandardError => ex
+              if enable_delayed_message
+                publish_to_delayed_message(delivery_info, metadata, payload)
+              else
+                channel.nack(delivery_info.delivery_tag)
+              end
+            ensure
+              channel.ack(delivery_info.delivery_tag)
             end
           end
 
+          def publish_to_delayed_message(delivery_info, metadata, payload)
+            headers = metadata[:headers] || {}
+            headers["x-delay"] = 1000
+            delayed_message_exchange.publish(payload, { :headers => headers })
+            channel.ack(delivery_info.delivery_tag)
+          end
+
+          def delayed_message_exchange
```

Comment:
I tried this, but kept getting timeouts when passing along the exchange and queue.  I was never able to figure out why, but there wasn't any particular reason that I could ascertain for why the timeout was happening.